UnitsNet/UnitConverter.cs

```diff
@@ -20,30 +20,124 @@
 // THE SOFTWARE.
 
 using System;
+using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
 using System.Reflection;
 using UnitsNet.InternalHelpers;
 using UnitsNet.Units;
+
 #if WINDOWS_UWP
-using Culture = System.String;
-using FromValue = System.Double;
+    using FromValue = System.Double;
 #else
-using Culture = System.IFormatProvider;
-using FromValue = UnitsNet.QuantityValue;
+    using FromValue = UnitsNet.QuantityValue;
 #endif
 
 namespace UnitsNet
 {
+    using ConversionFunctionLookup = ValueTuple<Type, Enum, Type, Enum>;
+
+    public delegate IQuantity ConversionFunction(IQuantity inputValue);
+
     /// <summary>
     ///     Convert between units of a quantity, such as converting from meters to centimeters of a given length.
     /// </summary>
-    public static class UnitConverter
+    public sealed partial class UnitConverter
     {
+        public static UnitConverter Default { get; }
+
         private static readonly string QuantityNamespace = typeof(Length).Namespace;
         private static readonly string UnitTypeNamespace = typeof(LengthUnit).Namespace;
         private static readonly Assembly UnitsNetAssembly = typeof(Length).GetAssembly();
 
+        private Dictionary<ConversionFunctionLookup, ConversionFunction> conversionFunctions;
+
+        static UnitConverter()
+        {
+            Default = new UnitConverter();
+
+            RegisterDefaultConversions(Default);
+        }
+
+        public UnitConverter()
+        {
+            conversionFunctions = new Dictionary<ConversionFunctionLookup, ConversionFunction>();
```

Comment:
Can initialize this on field and also make field readonly. Also use `_` prefix on private instance fields.